zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceTest.java

```diff
@@ -1024,6 +1026,71 @@ public void shouldWriteMigratedEventForLocalVariable() {
         .hasTenantId(variable2.getTenantId());
   }
 
+  @Test
+  public void shouldUnsubscribeUnmappedMessageBoundaryEvent() {
+    // given
+    final String processId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "2";
+
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .userTask("A")
+                    .boundaryEvent("boundary")
+                    .message(m -> m.name("message").zeebeCorrelationKeyExpression("key"))
+                    .endEvent()
+                    .moveToActivity("A")
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .userTask("B")
```

Comment:
💭 Previously we had an [issue](https://github.com/camunda/zeebe/issues/15570) while testing message subscription creation while migrating tasks with same ids. I think, it is not needed for this PR but we should consider writing a test case (with task ids not changing) for deleting and recreation combined when there are no mapping specified for boundary events.

Reply:
Yes, let's do that for deletion and recreation